### ISSUE TITLE
Bminer crashes in the background

### ISSUE THREAD

**Issue by boro2000** [MultiPoolMiner/MultiPoolMiner#1306]
Hi,
I have got big problems with the bminer crashing in the background in MPM 2.7.2.
These crashes only happen, when using the v2.7.2 Start-SubProcess method. It looks like bminer does not like being started like this, because, using the old v2.7.1 Start-SubProcess method, it runs stable for days. 

There is no error message in the bminer*.txt logfile.
I reduced the test setup to only two miners (bminer and dstm) + I added Write-Log commands to the Miner-methods GetStatus and SetStatus, to see what happens.

```
2018-02-23 23:43:15 INFO: Updating exchange rates from Coinbase. 
2018-02-23 23:43:15 INFO: Loading saved statistics. 
2018-02-23 23:43:18 INFO: Loading pool information. 
2018-02-23 23:43:19 INFO: Selecting best pool for each algorithm. 
2018-02-23 23:43:19 INFO: Getting miner information. 
2018-02-23 23:43:27 INFO: Calculating profit for each miner. 
2018-02-23 23:43:32 INFO: Bminer::GetStatus() return this.Status=Idle
2018-02-23 23:43:32 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:43:32 INFO: Bminer::GetStatus() return this.Status=Idle
2018-02-23 23:43:33 INFO: Bminer::GetStatus() return this.Status=Idle
2018-02-23 23:43:33 INFO: Starting miner (Bminer): 'C:\Users\...\mpm\Bin\Equihash-BMiner\bminer.exe -api 127.0.0.1:1880 -uri stratum://...@equihash.usa.nicehash.com:3357 -nofee'
2018-02-23 23:43:33 INFO: Bminer::GetStatus() return this.Status=Idle
2018-02-23 23:43:33 INFO: Bminer::SetStatus(Running) case Running this.Status=Idle
2018-02-23 23:43:33 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:43:33 INFO: Start waiting before next run. 
2018-02-23 23:44:15 INFO: Finish waiting before next run. 
2018-02-23 23:44:15 INFO: Saving hash rates. 
2018-02-23 23:44:15 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Bminer: When Crypto-mining Made Fast (v5.4.0-ae18e12) 
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Watchdog has started.                        
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Connected to equihash.usa.nicehash.com:3357  
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Starting miner on device 0...                
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Started miner on device 0                    
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Starting miner on device 1...                
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Subscribed to stratum server                 
2018-02-23 23:44:53 VERBOSE: Bminer: [INFO] [2018-02-23T23:43:36+01:00] Set nonce to 3d00ad6e82                      
... and so on ...
2018-02-23 23:44:53 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:44:54 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:44:54 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:44:54 INFO: Starting next run...
2018-02-23 23:44:54 INFO: Updating exchange rates from Coinbase. 
2018-02-23 23:44:55 INFO: Loading saved statistics. 
2018-02-23 23:44:57 INFO: Loading pool information. 
2018-02-23 23:44:58 INFO: Selecting best pool for each algorithm. 
2018-02-23 23:44:59 INFO: Getting miner information. 
2018-02-23 23:45:08 INFO: Calculating profit for each miner. 
2018-02-23 23:45:13 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:45:14 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:45:14 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:45:14 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:45:14 INFO: Bminer::GetStatus() return Running this.Status=Running
2018-02-23 23:45:17 INFO: Start waiting before next run. 
2018-02-23 23:45:57 INFO: Finish waiting before next run. 
2018-02-23 23:45:57 INFO: Saving hash rates. 
2018-02-23 23:45:57 INFO: Bminer::GetStatus() return Failed this.Status=Running
2018-02-23 23:45:57 INFO: Bminer::GetStatus() return Failed this.Status=Running
2018-02-23 23:45:57 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:45:57 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:45:57 INFO: Starting next run...
2018-02-23 23:45:57 INFO: Updating exchange rates from Coinbase. 
2018-02-23 23:45:58 INFO: Loading saved statistics. 
2018-02-23 23:46:00 INFO: Loading pool information. 
2018-02-23 23:46:01 INFO: Selecting best pool for each algorithm. 
2018-02-23 23:46:01 INFO: Getting miner information. 
2018-02-23 23:46:09 INFO: Calculating profit for each miner. 
2018-02-23 23:46:13 INFO: Bminer::GetStatus() return Failed this.Status=Running
2018-02-23 23:46:13 INFO: Dstm::GetStatus() return this.Status=Idle
2018-02-23 23:46:14 INFO: Starting miner (ExcavatorNvidia2): 'C:\Users\...\mpm\Bin\Excavator\excavator.exe -p 23456 -c NiceHash_Equihash_..._2_Nvidia.json -na'
2018-02-23 23:46:14 INFO: ExcavatorNvidia2::GetStatus() return this.Status=Idle
2018-02-23 23:46:14 INFO: ExcavatorNvidia2::SetStatus(Running) case Running this.Status=Idle
2018-02-23 23:46:14 INFO: ExcavatorNvidia2::GetStatus() return Running this.Status=Running
2018-02-23 23:46:14 INFO: Bminer::GetStatus() return Failed this.Status=Running
2018-02-23 23:46:14 INFO: Start waiting before next run. 

```

**Comment by boro2000** [comment]
What is a little special about bminer.exe is, that this miner seems to start a second bminer.exe process. The first process stays put and idles, the second bminer.exe process seems to do the math, as it uses cpu.
![image](https://user-images.githubusercontent.com/28839768/36621397-22704e38-18f8-11e8-84e9-1473cd0f81c2.png)

**Comment by borzaka** [comment]
Be careful with bminer. It's high probability that it's scam miner, and mines way more to the developer and reporting hore hashes on the client side.

See:
#1127
#1242

**Comment by boro2000** [comment]
Thank you, for the warning.
Actually, I am not so sure, that it is a scam. I have had it running in parallel to Dstm on a six-identical-1070 GPU rig. Every miner got 3 GPU, both were mining to Nicehash - after having monitored for five days, Bminer (using -nofee) option produced approx. 3% more income than Dstm. Also, I could not find undercover mining, up to now (which does not meen, that there is any). The thing is, Bminer contacts it's maker regularly. I suspect, the author plans to sell some statistical or monitoring capabilities, later.

**Comment by BobTheTool** [comment]
Hi, same here. Bminer run again 49 times in 50 min because it crash at every run...

**Comment by UselessGuru** [comment]
> Hi, same here. Bminer run again 49 times in 50 min because it crash at every run...

Solution: remove bminer.ps1 (it is not really faster)

We know that MPM is calling Bminer properly (for some people it works well), and it also works when run 'standalone' using the exact same parameters.
So there is another reason for the crashes. I have the suspicion that it could be caused by windows itself. PalginHRS hack had a similar issue:
https://bitcointalk.org/index.php?topic=2765610.msg29189721#msg29189721

So what version of Windows are you using?

**Comment by UselessGuru** [comment]
@aaronsace 

Can you please follow up on this one - I have not had a chance to dive into this bit of code (all new stuff again :-))

**Comment by romajc84** [comment]
Bminer works great on my 1GPU PC.  But on my 2x7GPU rigs it does not.

**Comment by aaronsace** [comment]
@boro2000 Yes, that's what I thought it was but the change you suggested makes it less safe. I don't want to change that safety check as it's vital that hidden process definitely get killed no matter what. In the next version, I will add a process count setting on each miner. 

@boro2000 and @romajc84, is it 1 bminer process per GPU?

**Comment by boro2000** [comment]
@aaronsace 
it is just two bminer processes, no matter how many GPU.
Ok, looking forward this process counter. For now, what I do not understand, why the change will make it less safe. At the moment, I cannot imagine a situation, where that might oppose a problem.

**Comment by boro2000** [comment]
@aaronsace
.. a correction: if there is only one GPU, there is only one bminer process.

**Comment by UselessGuru** [comment]
Fixed with #1353
